documentation/jetty/modules/programming-guide/pages/security/openid-support.adoc

```diff
@@ -0,0 +1,92 @@
+This will give you a Client ID and Client Secret.
+Once set up you must also register all the possible URI's for your webapp with the path `/j_security_check` so that the OpenId Provider will allow redirection back to the webapp.
+
+These may look like
+
+ * `http://localhost:8080/openid-webapp/j_security_check`
+
+ *  `https://example.com/j_security_check`
+
+
+== Embedded Configuration
+
+=== Define the `OpenIdConfiguration` for a specific OpenID Provider.
+
+If the OpenID Provider allows metadata discovery then you can use.
+
+[,java,indent=0]
+----
+include::code:example$src/main/java/org/eclipse/jetty/docs/programming/security/OpenIdDocs.java[tags=createConfigurationWithDiscovery]
+----
+
+Otherwise, you can manually enter the necessary information:
+
+[,java,indent=0]
+----
+include::code:example$src/main/java/org/eclipse/jetty/docs/programming/security/OpenIdDocs.java[tags=createConfiguration]
+----
+
+=== Configuring an `OpenIdLoginService`
+[,java,indent=0]
+----
+include::code:example$src/main/java/org/eclipse/jetty/docs/programming/security/OpenIdDocs.java[tags=configureLoginService]
+----
+
+=== Configuring an `OpenIdAuthenticator` with `OpenIdConfiguration` and Error Page Redirect
+[,java,indent=0]
+----
+include::code:example$src/main/java/org/eclipse/jetty/docs/programming/security/OpenIdDocs.java[tags=configureAuthenticator]
+----
+
+=== Usage
+
+==== Claims and Access Token
+Claims about the user can be found using attributes on the session attribute `"org.eclipse.jetty.security.openid.claims"`, and the full response containing the OAuth 2.0 Access Token can be found with the session attribute `"org.eclipse.jetty.security.openid.response"`.
+
+Example:
+[,java,indent=0]
+----
+include::code:example$src/main/java/org/eclipse/jetty/docs/programming/security/OpenIdDocs.java[tags=configureAuthenticator]
+----
+
+== Scopes
+The OpenID scope is always used but additional scopes can be requested which can give you additional resources or privileges.
+For the Google OpenID Provider it can be useful to request the scopes `profile` and `email` which will give you additional user claims.
+
+Additional scopes can be requested through the `start.ini` or `start.d/openid.ini` files, or with `OpenIdConfiguration.addScopes(...);` in embedded code.
+
+== Authorization
+
+If security roles are required they can be configured through a wrapped `LoginService` which is deferred to for role information by the `OpenIdLoginService`.
+
+This can be configured in XML through `etc/openid-baseloginservice.xml` in the Distribution, or in embedded code using the constructor for the `OpenIdLoginService`.
+
+[,java,indent=0]
+----
+include::code:example$src/main/java/org/eclipse/jetty/docs/programming/security/OpenIdDocs.java[tags=wrappedLoginService]
+----
+
+When using authorization roles, the setting `authenticateNewUsers` becomes significant.
```

Comment:
What class has `authenticateNewUsers`? 
This is the programming guide, if this is important in the operations guide, then must be specified there.